README.md

```diff
@@ -58,7 +58,7 @@ The list below indicates all algorithms supported by liboqs, but not all those a
 
 <!--- OQS_TEMPLATE_FRAGMENT_LIST_SIGS_START -->
 - **CRYSTALS-Dilithium**: Dilithium2, Dilithium3, Dilithium5
-- **Falcon**: Falcon-512, Falcon-1024
+- **Falcon**: Falcon-512, Falcon-1024, Falcon-512 (PADDED), Falcon-1024 (PADDED)
```

Comment:
Is this naming prudent? It makes specifying this alg a bit of a hassle (escaping space and bracket if used on a command line; always applying CAPS lock). What about "Falcon-padded-512"/1024?

Reply:
Good point regarding the space... I went with all caps because that's how the formats are referred to in the reference documentation (https://falcon-sign.info/impl/falcon.h.html, starting at line 120). There are no uses of `strcmp` in `liboqs/src`, only `strcasecmp`, so I don't think the capitalization poses an issue. (I do see some `strcmp`s in the KAT test helpers, which I introduced—oops.)

How about `Falcon-512-PADDED` / `Falcon-1024-PADDED`? This is in line with `FrodoKEM-640-SHAKE` / `FrodoKEM-640-AES` / `FrodoKEM-1344-SHAKE` / etc. I'd prefer to have the format modifier at the end, as this really is fundamentally the same algorithm as Falcon-512/1024.

This space/paren/caps convention has already made its way into PQClean, but I don't imagine it will be difficult to amend.

Reply:
Hmm... If you are looking for smth case-insensitive using `strcasecmp` inside liboqs, you should beware of the TR locale. `Dilithium` with its several `i`s is especially in danger. In OpenSSL we had to introduce `OPENSSL_strcasecmp` to deal with it.

Reply:
> I went with all caps because that's how the formats are referred to in the reference documentation (https://falcon-sign.info/impl/falcon.h.html, starting at line 120)

This argument is somewhat weak as it otherwise begs the question why we don't have "COMPRESSED" in the (original alg's) name, too. And No: I don't argue for that!

> How about Falcon-512-PADDED / Falcon-1024-PADDED? This is in line with FrodoKEM-640-SHAKE / FrodoKEM-640-AES / FrodoKEM-1344-SHAKE / etc. I'd prefer to have the format modifier at the end, as this really is fundamentally the same algorithm as Falcon-512/1024.

Yup -- I always hated the Frodo naming :-) Superfluous "KEM" and (to me) unlogical: Most Algs have a "base name", possibly a "variant modifier" and then a "strength indicator". That said, Sphincs also deviates from this, so why not another one? :-/ My argument btw does not contradict yours: falcon remains the basic algorithm, so falcon-padded is just a variant -- in two strength versions.

Reply:
> > I went with all caps because that's how the formats are referred to in the reference documentation (https://falcon-sign.info/impl/falcon.h.html, starting at line 120)
> 
> This argument is somewhat weak as it otherwise begs the question why we don't have "COMPRESSED" in the (original alg's) name, too. And No: I don't argue for that!
>

For argument's sake, "COMPRESSED" is the default format, so it's implied.

> > How about Falcon-512-PADDED / Falcon-1024-PADDED? This is in line with FrodoKEM-640-SHAKE / FrodoKEM-640-AES / FrodoKEM-1344-SHAKE / etc. I'd prefer to have the format modifier at the end, as this really is fundamentally the same algorithm as Falcon-512/1024.
> 
> Yup -- I always hated the Frodo naming :-) Superfluous "KEM" and (to me) unlogical: Most Algs have a "base name", possibly a "variant modifier" and then a "strength indicator". That said, Sphincs also deviates from this, so why not another one? :-/ My argument btw does not contradict yours: falcon remains the basic algorithm, so falcon-padded is just a variant -- in two strength versions.

The naming (presumably) has more of an impact downstream, and that's your department, so I'll concede. :stuck_out_tongue_winking_eye: `Falcon-padded-512` and `Falcon-padded-1024` it is. At any rate, this will be a moot point in the near future (hopefully) when Falcon becomes NTRU-DSA or whatever name NIST settles on for the standardized version.

Reply:
@baentsch Following up on this after looking into patching in an updated name:

The `Falcon-* (PADDED)` name has already landed in PQClean. (I should probably have solicited feedback on that choice, earlier, mea culpa. :grimacing:) It should be an easy and quick fix to patch it (in `liboqs`) to `Falcon-*-padded` or `Falcon-*-PADDED` or what-have-you, in alg-param-variant order. However, switching to alg-variant-param will change the PQClean algorithm namespace from `PQCLEAN_FALCON512PADDED_*` to `PQCLEAN_FALCONPADDED512_*`. That's too big a change to fix easily with a patch, so it would require a PR upstream (one with a very large, albeit inconsequential diff) to change the name.

I'm OK with doing the work, but since we're aiming to cut a release candidate tomorrow, I thought I should check to see if it's worth potentially delaying that for naming conventions.

Reply:
Sorry in turn for having brought this up so late in the game. I might have compromised on the place where "padded" is placed, but the space and brackets are no-gos in my opinion: If we'd do a release with that choice, this will come and haunt us (as a release by definition is a commitment).

So indeed, two choices:
1) create a huge patch to enable running `copy-from-upstream` on the release with the "logical" name, removing that when upstream changed.
2) First fix upstream and delay release until that happened. 

1) is superfluous work which I'd hate; so how much time are we talking about on 2?

Reply:
Update: I just added said patch and re-ran `copy_from_upstream`.

Reply:
> The space and brackets are an easy fix; that doesn't change the namespace. I can add a very small patch to change Falcon-512 (PADDED) to Falcon-512-padded in the api.h and META.yml files, and nothing else will need to be done.

Thanks for that change. But this would still mean that we'd have to keep supporting an unlogical name :-( I still wouldn't if we don't release 0.10.0 right away. And [the tracking sheet](https://github.com/open-quantum-safe/liboqs/milestone/23) doesn't let me think we will...

If the upstream change is taking too long, what about also just changing "pretty_name_full" in [copy_from_upstream.yml](https://github.com/open-quantum-safe/liboqs/blob/b952262ad473cfb464cb6f300a80e9ac10fe96eb/scripts/copy_from_upstream/copy_from_upstream.yml#L231) (leaving "pqclean_scheme" unchanged until the naming gets "logic-fied" there too)?

Reply:
I went ahead and made the upstream PR: https://github.com/PQClean/PQClean/pull/549. It hit a lot of files but was pretty quick with some help from `sed`. I'll wait for your comments there before hitting merge.

Reply:
> I'll wait for your comments there before hitting merge.
 
None (negative :) from my side. Worthwhile trying to run `copy_from_upstream` on that branch, just to test it here already? Would give someone with write access in PQClean time to review more deeply. Thanks for doing this, @SWilson4 -- I honestly didn't think my comment would touch upon so many files -- but as you said, `sed` to the rescue :-)